### PR TITLE
fix(cli): Json output for vuln ctr adheres to filter flags

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Fix json output to should work in combination with any filtering flags

- vuln ctr list-assessments vuln_container.go:196
- vuln ctr scan vuln_container.go:392
- vuln ctr scan-status vuln_container.go:435
- vuln ctr show-assessment vuln_container.go:482

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: cli/cmd/vulnerability.go [hunk 1]
```diff
@@ -324,6 +314,55 @@ func pollScanStatus(requestID string) error {
...
+	}
+	return nil
+}
+
+type vulnerabilityDetailsReport struct {
+	VulnerabilityDetails filteredImageTable
+	Packages             filteredPackageTable
+}
+
+func buildReportTable(summary string, details string) string {
```
Comment (afiune) [inline cli/cmd/vulnerability.go:349]: same here, `buildReportTable()` is too generic.

**Review by afiune — CHANGES_REQUESTED** [review]

File: cli/cmd/vuln_container.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ Then navigate to Settings > Integrations > Container Registry.
 				return cli.OutputJSON(registries)
 			}
 
-			rows := [][]string{}
+			var rows [][]string
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply cli/cmd/vuln_container.go:126]: IDE flagged it as a suggested change. I see it listed as the preferred way on stackoverflow to use var when declaring without initialising. Makes sense to me, we don't need to initialise rows so why use the initialiser shorthand.  I don't see an 'official' Go statement on this practice. I'm curious to know where the ide spec comes from for this.

File: cli/cmd/vulnerability.go [hunk 3]
```diff
@@ -324,6 +314,55 @@ func pollScanStatus(requestID string) error {
...
 
+// Build the cli output for vuln ctr 'scan-status', 'scan' and 'show-assessment' commands
+func buildReports(assessment *api.VulnContainerAssessment) error {
+	summaryReport := buildVulnerabilitySummaryReportTable(assessment)
+	details := vulnerabilityDetailsReport{
+		VulnerabilityDetails: filterVulContainerImageLayers(assessment.Image),
+		Packages:             filterVulnContainerImagePackages(assessment.Image),
+	}
+	detailsReport := buildVulnerabilityDetailsReportTable(details)
+	filteredAssessment := assessment
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply cli/cmd/vulnerability.go:325]: I agree, will fix! Thanks

File: cli/cmd/vulnerability.go [hunk 4]
```diff
@@ -324,6 +314,55 @@ func pollScanStatus(requestID string) error {
 	}
 }
 
+// Build the cli output for vuln ctr 'scan-status', 'scan' and 'show-assessment' commands
+func buildReports(assessment *api.VulnContainerAssessment) error {
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply cli/cmd/vulnerability.go:318]: Good call, will fix!

File: cli/cmd/vulnerability.go (hunk 1 above)
Comment (dmurray-lacework) [reply cli/cmd/vulnerability.go:349]: Will fix this one too, thanks!

**Review by afiune — APPROVED** [review]
tenor-100765766

**Pull request merged**